### PR TITLE
[SC-109] Gnosis Chain Activation Spell

### PR DESCRIPTION
**Pull request by hexonaut**
Activating Gnosis Chain instance: https://forum.makerdao.com/t/proposal-for-activation-of-gnosis-chain-instance/22098

### CODE REVIEW

File: src/SparkTestBase.sol [hunk 1]
```diff
@@ -295,6 +296,18 @@ abstract contract SparkGoerliTestBase is SparkTestBase {
         executor = 0x4e847915D8a9f2Ab0cDf2FC2FD0A30428F25665d;
         domain = 'Goerli';
         poolAddressesProviderRegistry = IPoolAddressesProviderRegistry(0x1ad570fDEA255a3c1d8Cf56ec76ebA2b7bFDFfea);
+        disableE2E = true;          // TODO enable once this is fixed
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/SparkTestBase.sol]: Fixed.

File: src/SparkTestBase.sol [hunk 2]
```diff
@@ -82,8 +85,7 @@ abstract contract SparkTestBase is ProtocolV3TestBase {
                 pool
             );
 
-            // Goerli is broken so just disable for now
-            if (block.chainid != 5) {
+            if (!disableExportDiff) {
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/SparkTestBase.sol:88]: There is a loop though. I think it's messier to change.

File: src/proposals/20230927/SparkEthereum_20230927.sol [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import { SparkPayloadEthereum, Address } from '../../SparkPayloadEthereum.sol';
+
+interface IForwarder {
+    function execute(address payload) external;
+}
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/proposals/20230927/SparkEthereum_20230927.sol:8]: Usually we do not want to import dependencies, but the gov relay will be frequent enough that this makes sense. As this is just a cosmetic change let's make a note to do this after this PR so we don't have to redeploy the spell.

File: src/proposals/20230927/SparkEthereum_20230927.sol [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * @author Phoenix Labs
+ * @dev    This proposal activates the Gnosis Chain instance of Spark Lend
+ * Forum:  https://forum.makerdao.com/t/proposal-for-activation-of-gnosis-chain-instance/22098
+ * Vote:   https://vote.makerdao.com/polling/QmVcxd7J#poll-detail
+ */
+contract SparkEthereum_20230927 is SparkPayloadEthereum {
+
+    using Address for address;
+
+    address public constant GNOSIS_FORWARDER = 0x44f993EAe9a420Df9ffa5263c55f6C8eF46c0340;
```
[the remark this answers is not in the document]
Comment (hexonaut) [reply src/proposals/20230927/SparkEthereum_20230927.sol:21]: Good idea.

**Review by barrutko — APPROVED** [review]
LGTM
Notes for the future:
- We need to remove duplicated code and import from `spark-gov-relay`
- We need to create an address book contract for constants
- We need to fix the E2E test

**Review by lucas-manuel — DISMISSED** [review]
LGTM as well, just need to sort out E2E discrepancy

**Review by lucas-manuel — APPROVED** [review]

**Review by barrutko — APPROVED** [review]

**Pull request merged**